test/base/src/androidTest/kotlin/io/realm/test/shared/EmbeddedObjectTests.kt

```diff
@@ -0,0 +1,1161 @@
+    //             }
+    //         }
+    //     }
+    // }
+    //
+
+    // ALREADY COVERED In RealmSchemaTest
+    // @org.junit.Test
+    // fun realmObjectSchema_isEmbedded() {
+    //     Assert.assertTrue(realm.schema[EmbeddedSimpleChild.NAME]!!.isEmbedded)
+    //     Assert.assertFalse(realm.schema[AllTypes.CLASS_NAME]!!.isEmbedded)
+    // }
+    //
+
+    // NOT RELEVANT - Not specific to Dynamic realms, so above should be sufficient
+    // @org.junit.Test
+    // fun dynamicRealm_realmObjectSchema_isEmbedded() {
+    //     DynamicRealm.getInstance(realm.configuration).use { realm ->
+    //         Assert.assertTrue(realm.schema[EmbeddedSimpleChild.NAME]!!.isEmbedded)
+    //         Assert.assertFalse(realm.schema[AllTypes.CLASS_NAME]!!.isEmbedded)
+    //     }
+    // }
+    //
+
+    @Test
+    fun deleteParentObject_deletesEmbeddedChildren() {
+        val parent = EmbeddedParent().apply {
+            child = EmbeddedChild("child1")
+            childList.addAll(setOf(EmbeddedChild("child2"), EmbeddedChild("child3")))
+        }
+
+        val managedParent = realm.writeBlocking { copyToRealm(parent) }
+
+        assertEquals(3, realm.query<EmbeddedChild>().find().size)
+
+        realm.writeBlocking { findLatest(managedParent)!!.let { delete(it) } }
+
+        assertEquals(0, realm.query<EmbeddedChild>().find().size)
+    }
+
+    // @org.junit.Test
+    // fun dynamicRealm_deleteParentObject_deletesEmbeddedChildren() =
+    //     DynamicRealm.getInstance(realm.configuration).use { realm ->
+    //         realm.executeTransaction {
+    //             val parent = realm.createObject("EmbeddedSimpleParent", "parent")
+    //             Assert.assertEquals(0, realm.where("EmbeddedSimpleChild").count())
+    //
+    //             val child = realm.createEmbeddedObject("EmbeddedSimpleChild", parent, "child")
+    //             Assert.assertEquals(1, realm.where("EmbeddedSimpleChild").count())
+    //
+    //             parent.deleteFromRealm()
+    //             Assert.assertFalse(child.isValid)
+    //             Assert.assertEquals(0, realm.where("EmbeddedSimpleParent").count())
+    //             Assert.assertEquals(0, realm.where("EmbeddedSimpleChild").count())
+    //         }
+    //     }
+    //
+
+    @Test
+    fun deleteParentEmbeddedObject_deletesEmbeddedChildren() {
+        val parent = EmbeddedParent().apply {
+            child = EmbeddedChild("child1").apply { innerChild = EmbeddedInnerChild() }
+            childList.add(EmbeddedChild("child2").apply { innerChild = EmbeddedInnerChild() })
+        }
+
+        val managedParent = realm.writeBlocking { copyToRealm(parent) }
+
+        assertEquals(2, realm.query<EmbeddedChild>().find().size)
+        assertEquals(2, realm.query<EmbeddedInnerChild>().find().size)
+
+        realm.writeBlocking {
+            findLatest(managedParent)!!.run {
+                child = null
+                childList.clear()
```

Comment:
clear behaves like remove? I was expecting it to remove elements from the list not delete them from the Realm?

Reply:
Embedded Objects _must_ have a parent, so clearing the parent reference will implicitly delete them.